### PR TITLE
Logging of dealid and proposalID during deals, and reporting in finalized task

### PR DESCRIPTION
**Pull request by willscott**

### CODE REVIEW

File: tasks/retrieval_deal.go [hunk 1]
```diff
@@ -219,7 +219,35 @@ func (de *retrievalDealExecutor) executeAndMonitorDeal(ctx context.Context, upda
 				if event.Event == retrievalmarket.ClientEventDealAccepted {
 					stage = RetrievalStageDealAccepted
 					dealStage = RetrievalStages[stage]()
-					err := updateStage(ctx, stage, dealStage)
+					//there should be a deal now.
+					allRetrievals, err := de.node.ClientListRetrievals(ctx)
```
Comment (willscott) [inline tasks/retrieval_deal.go:223]: are there other things to match on here in pulling out the deal from the retrieval? cc @hannahhoward

**Comment by willscott** [comment]
fix #271

File: tasks/retrieval_deal.go [hunk 2]
```diff
@@ -219,7 +219,35 @@ func (de *retrievalDealExecutor) executeAndMonitorDeal(ctx context.Context, upda
...
+								break
+							}
+						}
+					}
+					if expectedID != 0 {
+						allDeals, err := de.node.ClientListDeals(ctx)
+						if err != nil {
+							return err
+						}
+						for _, maybeOurDeal := range allDeals {
```
Comment (hannahhoward) [inline tasks/retrieval_deal.go:244]: This isn't going to work -- these are not the same DealID. The DealID on storage side is an onchain incrementing integer for all deals in the network. The DealID for retrieval is different -- it's just an off chain incrementing integer per client that has nothing to do with the storage deal id.

File: tasks/retrieval_deal.go [hunk 3]
```diff
@@ -219,7 +219,35 @@ func (de *retrievalDealExecutor) executeAndMonitorDeal(ctx context.Context, upda
...
+					//there should be a deal now.
+					allRetrievals, err := de.node.ClientListRetrievals(ctx)
+					expectedPayload, parseErr := cid.Parse(de.task.PayloadCID)
+					if parseErr != nil {
+						return err
+					}
+					expectedID := abi.DealID(0)
+					if err == nil {
+						for _, maybeOurRetrieval := range allRetrievals {
+							if maybeOurRetrieval.PayloadCID.Equals(expectedPayload) {
```
Comment (hannahhoward) [inline tasks/retrieval_deal.go:231]: Miner ID Would be another thing we can check -- that's what we use above.

File: tasks/retrieval_deal.go (hunk 2 above)
Comment (hannahhoward) [reply tasks/retrieval_deal.go:244]: There's not a really great correlation here -- but we could use payload CID + miner ID.... except that can definitely be duplicated. Our best piece of information is what we have locally when we create the retrieval from the storage deal -- when we create the retrieval deal after a storage, we should log the proposal CID.

File: tasks/storage_deal.go [hunk 4]
```diff
@@ -379,7 +384,14 @@ func (de *storageDealExecutor) executeAndMonitorDeal(ctx context.Context, update
 			}
 
 			if len(info.DealStages.Stages) > 0 {
-				err = updateStage(ctx, storagemarket.DealStates[info.State], toStageDetails(info.DealStages.Stages[len(info.DealStages.Stages)-1]))
+				newState := storagemarket.DealStates[info.State]
```
Comment (hannahhoward) [inline tasks/storage_deal.go:387]: This also it not going to work. We don't get the DealID till the deal is on chain unfortunately, so we have to wait till at least the publish point to get the deal ID

**Review by hannahhoward — CHANGES_REQUESTED** [review]
Most of this code is not actually going to work. I'll reach out 1-on-1 to see if we can get the best approach here.

**Review by hannahhoward — APPROVED** [review]

**Pull request merged**